src/main/java/astvisitors/JasminCodeGeneratorVisitor.java

```
@@ -0,0 +1,684 @@
+        	case INT: case BOOL:
+        		appendLine("ineg");
+        		break;
+        	case DOUBLE:
+        		appendLine("dneg");
+        		break;
+        	}
+        }
+        else {
+        	generateCodeForLogicalNegation();
+        }
+
+    }
+
+    @Override
+    public void visit(LogicalOrExprNode node) {
+        node.getLeftNode().accept(this);
+        node.getRightNode().accept(this);
+        generateCodeForLogicalOr();
+    }
+
+    @Override
+    public void visit(PrintNode node) {
+    	incrementStack();
+    	appendLine("getstatic java/lang/System/out Ljava/io/PrintStream;");
+        for (AExpr child : node.getParameters()) {
+            child.accept(this);
+        }
+        appendLine("invokevirtual java/io/PrintStream/println(Ljava/lang/String;)V");
+        decrementStack();
+        decrementStack();
+    }
+
+    @Override
+    public void visit(CustomFuncCallStmtNode node) {
+        for (AExpr child : node.getParameters()) {
+            child.accept(this);
+        }
+        
+        append("invokestatic program/");
+        generateFunctionSignature(node);
+        if(!currentLocalFunctions.contains(node.getId())) {
+        	currentLocalFunctions.add(node.getId());
+        	maxLocals = Math.max(maxLocals, currentLocalFunctions.size() + currentVariableEnvironment.size());
+        }
+    }
+
+	@Override
+	public void visit(IntegerCastNode node) {
+		// TODO Auto-generated method stub
+		
+	}
+
+	@Override
+	public void visit(DoubleCastNode node) {
+		// TODO Auto-generated method stub
+		
+	}
+	
+	private void appendLine(String s) {
+		sb.append(s).append('\n');
+	}
+	private void append(String s) {
+		sb.append(s);
+	}
+	private int getNextLabel() {
+		return labelCounter++;
+	}
+	private void incrementStack() {
+		currentStackSize++;
+		if(currentStackSize > maxStackSize) {
+			maxStackSize = currentStackSize;
+		}
+	}
+	private void decrementStack() {
+		currentStackSize--;
+		if (currentStackSize < 0) {
```

Should this happen? Why not just let it blow up in our tests?